### PR TITLE
Average by group

### PR DESCRIPTION
**Pull request by stuart-cls**
Implements #231 

- [x] Add a few more other_var methods and matching GUI/tests
- [x] documentation :)

### CODE REVIEW

File: orangecontrib/spectroscopy/widgets/owaverage.py [hunk 1]
```diff
@@ -67,10 +67,35 @@ def set_data(self, dataset):
         self.commit()
 
     @staticmethod
-    def average_table(table):
+    def average_table(table, other_vars="None"):
```
Comment (markotoplak) [inline orangecontrib/spectroscopy/widgets/owaverage.py]: Wouldn't it be enough to just do "Mean" and not complicate with options. This one keeps the most information. What do you think?

Comment (stuart-cls) [reply orangecontrib/spectroscopy/widgets/owaverage.py]: Yes, great suggestion. It's easy to get carried away...

**Comment by stuart-cls** [comment]
I think this is ready to go. Two potential gotchas:

1. #234 might be a problem with the output OWAverage is generating, although I don't see how.
2. I couldn't quite get the documentation to work (when I press F1). Maybe we can add a small blurb somewhere or at least  a link to how Orange does it?

**Comment by markotoplak** [comment]
Thank you, I like what you did and how you did it. 

The bug you found is independent of this PR: if I save and open the averaged data I still get the error.

You need to build the documentation to see it:

    cd doc
    make htmlhelp

**Pull request merged**